src/renderer/components/dock/log-resource-selector.tsx

```diff
@@ -21,93 +21,85 @@
 
 import "./log-resource-selector.scss";
 
-import React, { useEffect } from "react";
+import React from "react";
 import { observer } from "mobx-react";
 
-import { Pod } from "../../../common/k8s-api/endpoints";
+import type { Pod } from "../../../common/k8s-api/endpoints";
 import { Badge } from "../badge";
-import { Select, SelectOption } from "../select";
+import { GroupSelectOption, Select, SelectOption } from "../select";
 import { LogTabData, logTabStore } from "./log-tab.store";
-import { podsStore } from "../+workloads-pods/pods.store";
 import type { TabId } from "./dock.store";
 
-interface Props {
-  tabId: TabId
-  tabData: LogTabData
-  save: (data: Partial<LogTabData>) => void
-  reload: () => void
+export interface LogResourceSelectorStore {
+  changeSelectedPod(tabId: TabId, newSelectedPod: Pod): void;
+  mergeData(tabId: TabId, data: Partial<LogTabData>): void;
 }
 
-export const LogResourceSelector = observer((props: Props) => {
-  const { tabData, save, reload, tabId } = props;
-  const { selectedPod, selectedContainer, pods } = tabData;
-  const pod = new Pod(selectedPod);
-  const containers = pod.getContainers();
-  const initContainers = pod.getInitContainers();
-
-  const onContainerChange = (option: SelectOption) => {
-    save({
-      selectedContainer: containers
-        .concat(initContainers)
-        .find(container => container.name === option.value),
-    });
-    reload();
-  };
+export interface LogResourceSelectorProps {
+  tabId: TabId;
+  pod: Pod;
+  pods: Pod[];
+  selectedContainer: string;
+  store?: LogResourceSelectorStore;
+}
 
-  const onPodChange = (option: SelectOption) => {
-    const selectedPod = podsStore.getByName(option.value, pod.getNs());
+function getSelected<T>(groups: GroupSelectOption<SelectOption<T>>[], value: T): SelectOption<T> | undefined {
+  for (const group of groups) {
+    for (const option of group.options) {
+      if (option.value === value) {
+        return option;
+      }
+    }
+  }
```

Comment:
I guess previous option for setting `Select` value was shorter and more explicit?

We don't need to iterate over available group, we can just set an "arbitrary" value and the React Select will do search itself.

```
value={{ label: pod.getName(), value: pod.getName() }}
```

Reply:
I tried setting it to just `pod.getName()` hoping that it would pick up the `value` correctly. But that didn't work. Will revert.